nova_vm/src/execution/environments/declarative_environment.rs

```diff
@@ -0,0 +1,29 @@
+use super::Environment;
+use crate::types::Value;
+use std::collections::HashMap;
+
+/// 9.1.1.1 Declarative Environment Records
+/// https://tc39.es/ecma262/#sec-declarative-environment-records
+#[derive(Debug)]
+pub struct DeclarativeEnvironment {
+    pub outer_env: Option<Environment>,
+    pub bindings: HashMap<&'static str, Binding>,
```

Comment:
Nice suggestion. I had added the `&'static str` to avoid lifetimes while copying the spec entities but this would be better. Will work on implementing this.

Reply:
The `ParallelVec`s in question should be sorted, and all code should appropriately use logarithmic lookups (stdlib has us covered here with slice methods).

If we're to do away with `HashMap`s entirely for some parts of the code\*, we should have a generic type built on `ParallelVec` for reuse.

\* There are other hash map types and algorithms and allocator configurations? Maybe those would be more suitible? A `HashMap` in an arena allocator for example. I believe there are already libraries for this.

Reply:
Or BTrees? https://doc.rust-lang.org/std/collections/struct.BTreeMap.html